ush/python/pygfs/task/aero_prepobs.py

```diff
@@ -0,0 +1,243 @@
+
+    @logit(logger)
+    def initialize(self) -> None:
+        """
+        List needed raw obs files.
+        Copy the raw obs files to $DATA/obs.
+        Link over the needed executable.
+        Generate corrosponding YMAL file.
+        Run IODA converter.
+        """
+        self.task_config.COM_OBS_CHEM = os.path.join(self.task_config.COM_OBS, 'chem')
+        if os.path.exists(self.task_config.COM_OBS_CHEM):
+            rmdir(self.task_config.COM_OBS_CHEM)
+        FileHandler({'mkdir': [self.task_config.COM_OBS_CHEM]}).sync()
+
+        self.task_config.DATA_OBS = os.path.join(self.task_config.DATA, 'obs')
+        if os.path.exists(self.task_config.DATA_OBS):
+            rmdir(self.task_config.DATA_OBS)
+        FileHandler({'mkdir': [self.task_config.DATA_OBS]}).sync()
+
+        self.task_config.prepaero_yaml = []
+        for sensor in self.task_config.sensors:
+            raw_files = self.list_raw_files(sensor)
+            self.task_config.input_files = self.copy_obs(raw_files)
+            self.link_obsconvexe()
+            self.task_config.prepaero_config = self.get_obsproc_config(sensor)
+
+            # generate converter YAML file
+            template = f"{self.runtime_config.CDUMP}.t{self.runtime_config['cyc']:02d}z.prepaero_viirs_{sensor}.yaml"
+            _prepaero_yaml = os.path.join(self.runtime_config.DATA, template)
+            self.task_config.prepaero_yaml.append(_prepaero_yaml)
+            logger.debug(f"Generate PrepAeroObs YAML file: {_prepaero_yaml}")
+            save_as_yaml(self.task_config.prepaero_config, _prepaero_yaml)
+            logger.info(f"Wrote PrepAeroObs YAML to: {_prepaero_yaml}")
+
+    @logit(logger)
+    def list_raw_files(self, sensor) -> List[str]:
+        """
+        List all files in the predefined directory that match the predefined sensor and within the time window.
+        """
+        if sensor == 'n20':
+            sensor = 'j01'
+        dir1 = f"{self.task_config.data_dir}/{datetime_to_YMD(self.task_config.window_begin)}"
+        dir2 = f"{self.task_config.data_dir}/{datetime_to_YMD(self.task_config.window_end)}"
+
+        if dir1 == dir2:
+            files = os.listdir(dir1)
+            allfiles = [os.path.join(dir1, file) for file in files]
+            allfiles.sort()
+        else:
+            files_1 = os.listdir(dir1)
+            allfiles_1 = [os.path.join(dir1, file) for file in files_1]
+            files_2 = os.listdir(dir2)
+            allfiles_2 = [os.path.join(dir2, file) for file in files_2]
+            allfiles = sorted(allfiles_1, allfiles_2)
+        matching_files = []
+        try:
+            for file in allfiles:
+                fshort = file.split('/')[-1].split('.')
+                yyyy = fshort[0][18:22]
+                mm = fshort[0][22:24]
+                dd = fshort[0][24:26]
+                HH = fshort[0][26:28]
+                MM = fshort[0][28:30]
```

Comment:
These are an interesting set of magic numbers in these slices. Can we use a regex to make this more robust?